File: laba1/lab1.cpp
#include <iostream>
using namespace std;
int main()
{   
    setlocale(LC_ALL,"RU");
    int digit, c=0, N, P;
    cout << "Enter number: ";
    cin >> N;
    cout << "Enter position: ";
    cin >> P;
    for(int i=N; i>0; i/=10)
    {
        digit = i%10; //последняя цифра
        c++;
        if(c==P){
            cout << "well done: " << digit << endl;
        }
    }
    return 0;
}

File: laba1/lab1.go
package main

import (
	"fmt"
)

func main() {
	var N int
	var P int
	var digit int
	var c int = 0

	fmt.Print("Введите число: ")
	fmt.Scan(&N)
	fmt.Print("Введите позицию: ")
	fmt.Scan(&P)

	if N < 0 || P < 0 {
		fmt.Println("Вы ввели неверное значение")
		return
	}

	for i := N; i > 0; i /= 10 {
		digit = i % 10
		c++
		if c == P {
			fmt.Println("well done:", digit)
		}
	}
}
